### PR TITLE
Deck fill out the rest of the commands with hotkeys and report formats

### PR DESCRIPTION
**Pull request by Cattlesquat**
Additional fields for deck:
* hotkeys to flip deck, turn it explicitly face up, turn it explicitly face down
* Different text field for "flip face down" from "flip face up"
* Menu text, hotkey, report format for "Save Deck"
* Menu text, hokey, report format for "Load Deck"

### CODE REVIEW

File: vassal-app/src/main/java/VASSAL/build/module/map/DrawPile.java [hunk 1]
```diff
@@ -751,6 +824,51 @@ else if (FACE_UP_MESSAGE.equals(key)) {
     else if (FACE_DOWN_MESSAGE.equals(key)) {
       dummy.setFaceDownMessage((String)value);
     }
+    else if (FACE_UP_HOTKEY.equals(key)) {
+      if (value instanceof String) {
+        value = NamedHotKeyConfigurer.decode((String) value);
+      }
+      dummy.setFaceUpKey((NamedKeyStroke) value);
```
Comment (uckelman) [inline vassal-app/src/main/java/VASSAL/build/module/map/DrawPile.java:831]: What's with all the assigning to `dummy`?

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/map/DrawPile.java:831]: That's just the way the Deck / Draw Pile unholy sandwich works. DrawPile is a game component and holds the fields and saves/loads them. Deck is the runtime element that's "sort of a GamePiece" and "sort of a Stack" and stuff.

File: vassal-app/src/main/java/VASSAL/build/module/map/MenuDisplayer.java [hunk 2]
```diff
@@ -138,6 +139,13 @@ public static JPopupMenu createPopup(GamePiece target, boolean global) {
       // Maps name to a list of commands with that name
       final HashMap<String, ArrayList<JMenuItem>> commandNames = new HashMap<>();
 
+      if (target instanceof Deck) {
+        if (c.length == 1) {
+          c[0].actionPerformed(new ActionEvent(popup, 0, ""));
+          return null;
+        }
+      }
+
```
Comment (BrentEaston) [inline vassal-app/src/main/java/VASSAL/build/module/map/MenuDisplayer.java:148]: What's this doing?

It looks like it Is automatically executing the menu option if there is only only one option in the menu of a Deck

**Review by BrentEaston — COMMENTED** [review]
Looks good overall, Just that query about the one-option menu handling

File: vassal-app/src/main/java/VASSAL/build/module/map/MenuDisplayer.java (hunk 2 above)
Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/map/MenuDisplayer.java:148]: Oh, yeah, that's what it does. Forgot about that. Seemed like "the right thing to do". Should I make it an option instead?

Comment (BrentEaston) [reply vassal-app/src/main/java/VASSAL/build/module/map/MenuDisplayer.java:148]: If you right-click on a Deck, I think the expectation is that you will be displayed a list of options to choose from, eve if it only one. What if you don't want to select that option? You can't get out of it, it just magically gets done.

Comment (Cattlesquat) [reply vassal-app/src/main/java/VASSAL/build/module/map/MenuDisplayer.java]: Ah, right good point. I was thinking of the "Draw Multiple" and "Draw Selected" kinds of things where it goes to another choose-or-cancel box anyway (and it's annoying/cumbersome to constantly have to pick from the menu of one thing and THEN choose-or-cancel). I have now restricted it to just those two possibilities.

**Comment by uckelman** [comment]
@BrentEaston Retest? Is this ok now?

**Review by BrentEaston — APPROVED** [review]

**Pull request merged**